### PR TITLE
[system-probe] Add `batchManager` to `http` package

### PR DESCRIPTION
**Pull request by p-lambert**
### What does this PR do?

* Add `http.batchManager` which is responsible for managing batch state between user and kernel space;
* Add `monitor.Sync()` method. This ensures all enqueued HTTP transactions in Kernel space are brought to Userspace;
* Add integration tests
* Fix eBPF program bugs
  * Fix buffer memset
  * Fix HTTP parsing for PUT requests
  * Remove unnecessary map lookup when flushing notifications

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Describe your test plan

Write there any instructions and details you may have to test your PR.

### CODE REVIEW

File: pkg/network/http/monitor.go [hunk 1]
```diff
@@ -90,70 +82,50 @@ func (http *Monitor) Start() error {
...
 
 	return nil
 }
 
+// Sync HTTP data between userspace and kernel space
+func (http *Monitor) Sync() {
+	reply := make(chan struct{}, 1)
+	defer close(reply)
+
+	// TODO: Add logic to ensure this won't deadlock during termination
```
Comment (ISauve) [inline pkg/network/http/monitor.go:124]: Maybe this should be resolved before merging?

**Review by ISauve — CHANGES_REQUESTED** [review]
We discussed this on slack but I'll leave a comment here too for anyone else who may take a look at this - I tested these changes locally and found some issues (tests were failing & my http requests weren't being reported), @p-lambert was able to confirm the issues on his end too

**Comment by p-lambert** [comment]
Sorry about that, @ISauve. Should be good now! Please give it another try

File: pkg/network/http/monitor.go (hunk 1 above)
Comment (p-lambert) [reply pkg/network/http/monitor.go]: Agreed! Done

**Review by ISauve — APPROVED** [review]
Tested it again & everything looks good! 🎉

**Review by derekwbrown — APPROVED** [review]

**Pull request merged**